### ISSUE TITLE
Losing user input when typing fast after backspace

### ISSUE THREAD

**Issue by Chris-Petty** [msupply-foundation/open-msupply#5035]
## What went wrong? 😲

If you press backspace and type relatively quickly (as a busy pharmacist might!), the state doesn't keep up and we end up deleting keystrokes after the backspaces. `356 +bsp +bsp + 2 + 5` may output `35` rather than `325`. This is bad as it risks wrong quantities prescribed OR inaccurate stock levels if the user doesn't notice the system messing up their valid input.

https://github.com/user-attachments/assets/e51fe2a4-7e62-4403-a51e-79613d446f42

Also the general jank form some state control flipping the values about after typing...

Note this might be even worse/easier to reproduce on a keyboard just as it's easier to be fast.

## Expected behaviour 🤔

No jank. Doesn't delete intended input.

## How to Reproduce 🔨

1. Prescription edit a line
2. have 3-4 digits
3. quickly press backspace 1 or more times and type in new digits - you may loose some of your input.

(Might be a requirement for the prescription to be backdated, or perhaps to have many transactions affecting the stock line you are attempting to issue)

## Your environment 🌱

- Open mSupply Version: 2.2.03 an 2.3.0-RC5
- Legacy mSupply Central Server Version: `main` 7.17.15 
- Platform: Samsung SM-X200
- Database type: sqlite

**Comment by lache-melvin** [comment]
Agree, when busy and trying to work quickly, this could easily result in errors - as well as general frustration! Setting to normal as can be worked around, if we engage the prescription line peacefully 😁 

Should be investigated how widespread this is though - is it the underlying input or a pattern in the prescription modal?

**Comment by andreievg** [comment]
Triage, just adding dispensing label to flag as something to look at in that bit of work. Can be quite frustrating to bruce lee

**Comment by DhanyaHerath** [comment]
Closing this to be looked at as part of https://github.com/msupply-foundation/open-msupply/issues/6188
